### DISCUSSION TITLE
Using GOLANG  to login users

### DISCUSSION THREAD

**Discussion by mariosplit** [pocketbase/pocketbase#3724] [Q&A]
I am trying to login a user from local users  collection using GOLANG but I keep getting error: 
Raw response body: {"code":404,"message":"Not Found.","data":{}}
Login response did not include expected token or user ID.

Here is my GO code: 
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

type LoginCredentials struct {
	Email    string `json:"email"`
	Password string `json:"password"`
}

type LoginResponse struct {
	Token string `json:"token"`
	User  struct {
		Id string `json:"id"`
	} `json:"user"`
	Error string `json:"error"`
}

func main() {
	url := "http://localhost:8090/api/v1/auth/login"
	credentials := LoginCredentials{
		Email:    "user@example.com",
		Password: "userpassword",
	}

	credsBytes, err := json.Marshal(credentials)
	if err != nil {
		panic(err)
	}

	resp, err := http.Post(url, "application/json", bytes.NewBuffer(credsBytes))
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		panic(err)
	}

	// Print out the raw response body for debugging
	fmt.Println("Raw response body:", string(body))

	var loginResponse LoginResponse
	if err := json.Unmarshal(body, &loginResponse); err != nil {
		panic(err)
	}

	if loginResponse.Error != "" {
		fmt.Printf("Error logging in: %s\n", loginResponse.Error)
		return
	}

	if loginResponse.Token == "" || loginResponse.User.Id == "" {
		fmt.Println("Login response did not include expected token or user ID.")
		return
	}

	fmt.Printf("Login successful! Token: %s, User ID: %s\n", loginResponse.Token, loginResponse.User.Id)
}
==========================
What am I doing wrong?

**Comment by ganigeorgiev** [comment]
The url is wrong. It should be `POST /api/collections/{collectionIdOrName}/auth-with-password`. For more details about the web API, please check https://pocketbase.io/docs/api-records/#auth-with-password.

**Reply by ganigeorgiev** [reply]
_Also note that `LoginResponse` struct doesn't look correct. Make sure to inspect the raw response and check the above link for the valid response fields._

**Reply by ganigeorgiev** [reply]
Also the body fields should be `identity` and `password` not `email` and `password`.

I don't know from where you copied the above code, but it is not correct. Please refer to the above API docs.

**Reply by mariosplit** [reply]
Your response is much appreciated. I am actually trying to figure out how the get a desktop (golang preferably) application as a front end and PocketBase as a backend.  Is that possible at all. At the moment I have a setup with Microsoft SharePoint 365 tables but PocketBase base would suit me much better.
